### PR TITLE
Refactored to match unix filter model

### LINKED ISSUE #723: ASGS not generating padcirc.slurm file?
Trying to get the latest ASGS/ASGSH running.  On initialize/hindcast, get this log message: 

`asgs_main.sh>submitJob()>hindcast : Submitting /projects/ncfs-dev/ncv99-nam-bob-2022/asgs23353/initialize/hindcast/padcirc.slurm.`

but jobErr contains:  `sbatch: error: Unable to open file padcirc.slurm`

and padcirc.slurm does not exist.   The usual log files (padcirc.out, adcirc.log, fort.16) are all empty.

### PR DESCRIPTION
**Pull request by jasonfleming**
I've refactored `qscript.pl` to simplify it and to make it match the unix filter model more closely. The long term goal here is to use the Template Toolkit, and I see this as a step in that direction.  

resolves #723

### CODE REVIEW

**Comment by jasonfleming** [comment]
Tested this successfully on my desktop machine. This is now running successfully on QBC and SuperMIC. Ready for review.

File: asgs_main.sh [hunk 1]
```diff
@@ -726,32 +726,96 @@ prepFile()
...
+                         > $qScriptResponse 2>> $SYSLOG
+   if [[ $? != 0 ]]; then
+      fatal "Failed to generate queue script."
+   fi
+   # extract queue script name from response
+   qscript=$(bashJSON.pl --key "qScriptFileName"        \
+                        < $qScriptResponse 2>> $SYSLOG)
+   # extract queue script from response
+   bashJSON.pl --key "script" < $qScriptResponse 2>> $SYSLOG \
+                              | base64 -d                    \
```
Comment (wwlwpd) [inline asgs_main.sh:801]: `base64` should be okay here, it is in GNU coreutils (had to look it up) - made me think to say, we just need to make sure any utils we use have a near 100% likelihood of being present. Stuff like `sed`, `awk`, etc are easy to assume.  Anyway, just wanted to point that out.

**Review by wwlwpd — APPROVED** [review]
Overall looks good, added a couple of comments for posterity's sake.

**Comment by wwlwpd** [comment]
Manually squashed the final merge of branch

**Pull request merged**